komorebi/src/animation_manager.rs

```diff
@@ -0,0 +1,76 @@
+use std::collections::HashMap;
+
+#[derive(Debug, Clone, Copy)]
+struct AnimationState {
+    pub in_progress: bool,
+    pub is_cancelled: bool,
+}
+
+#[derive(Debug)]
+pub struct AnimationManager {
+    animations: HashMap<isize, AnimationState>,
+}
+
+impl AnimationManager {
+    pub fn new() -> Self {
+        Self {
+            animations: HashMap::new(),
+        }
+    }
+
+    pub fn is_cancelled(&self, hwnd: isize) -> bool {
+        if !self.animations.contains_key(&hwnd) {
+            return false;
+        }
+
+        self.animations.get(&hwnd).unwrap().is_cancelled
```

Comment:
Is there a reasonable assumption that we can make about these boolean returns if the hwnd is not in the HashMap? It would be nice to default to something sensible instead of having a panic which kills the thread.

Reply:
Oh wait, I commented too soon  😅 

Maybe we can do this to avoid the unwraps:

```rust
if let Some(animation_state) = self.animations.get(&hwnd) { 
    animation_state.is_cancelled
} else {
   false
}
```

Reply:
Fixed that in new commit